winml/lib/Api/LearningModelSession.cpp

```diff
@@ -362,15 +259,16 @@ LearningModelSession::Run(
   // TODO : set the run_options
   onnxruntime::RunOptions run_options;
 
-  auto& io_binding = GetIOBinding(binding_impl, model_);
+  com_ptr<_winmla::IIOBinding> io_binding;
+  io_binding.attach(GetIOBinding(binding_impl, model_));
 
   // Invoke run on the ORT session.
-  WINML_THROW_IF_NOT_OK(inference_session_->Run(run_options, io_binding));
+  WINML_THROW_IF_FAILED(inference_session_->Run(&run_options, io_binding.get()));
```

Comment:
>WINML_THROW_IF_FAILED [](start = 2, length = 21)

NOT_OK is not the same as FAILED - why this change?

Reply:
nope.   not the same.  NOT_OK is for status.IsOK() checks.    This moved from an InferenceSession (ort type) to a IInferenceSession (com_ptr) and uses HRESULTS now across the ABI.    So it had to change from NOT_OK to FAILED.    

---
In reply to: [347042739](https://github.com/microsoft/onnxruntime/pull/2406#discussion_r347042739) [](ancestors = 347042739)